### PR TITLE
Filter out unsubscribed documents key in DocEvent 

### PR DESCRIPTION
**Pull request by chacha912**
**What this PR does / why we need it**:

image 

When firing a DocEvent, instead of directly passing the documentKeys of the published event, 
it is passed to the subscribed documentKey.
Previously, an error occurred in the following code when a client that only subscribed to `docKey1`
received a documentKeys of [`docKey1`, `docKey2`].

```go
func (c *Client) Watch(
  ctx context.Context,
  docs ...*document.Document,
) (<-chan WatchResponse, error) {
  // ...
  // handle WatchDocumentsResponse
  handleResponse := func(pbResp *api.WatchDocumentsResponse) (*WatchResponse, error) {
    switch resp := pbResp.Body.(type) {
    case *api.WatchDocumentsResponse_Event:
      eventType, err := converter.FromEventType(resp.Event.Type)

      switch eventType {
      case types.DocumentsWatchedEvent, types.DocumentsUnwatchedEvent, types.PresenceChangedEvent:
        for _, k := range converter.FromDocumentKeys(resp.Event.DocumentKeys) {
           cli, err := converter.FromClient(resp.Event.Publisher)
           // 💣 The document Keys in WatchDocumentsResponse may contain document keys 
           // that the client is not subscribed to.  
           attachment := c.attachments[k.String()]				 
     }
}
```

**Which issue(s) this PR fixes**:

Address https://github.com/yorkie-team/yorkie-js-sdk/pull/455

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

**Comment by hackerwins** [comment]
As we discussed, I think it would be nice to add test code like below.

1. `clientA` attaches `doc1`
2. `clientB` attaches `doc1`
3. `clientA` attaches `doc2`. At this time, checks whether information of `doc2` is coming to `clientB`

**Comment by chacha912** [comment]
@hackerwins Thank you for suggesting a testing procedure. 
I added the test as follows:

1. `Client A` attaches `doc1`
2. `Client A` watches `doc1`
3. `Client B` attaches `doc1` and `doc2`
4. `Client B` watches `doc1` and `doc2`. <- At this point, check whether `Client A` receive the information of `doc2`

In the original logic, an error occurs at step 4, but the test is passed after modification.
panic runtime

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.
I left a trivial comment.

File: test/integration/peer_awareness_test.go [hunk 1]
```diff
@@ -122,4 +122,83 @@ func TestPeerAwareness(t *testing.T) {
...
+		assert.NoError(t, err)
+
+		// 02. PeersChanged is triggered when another client closes the watch
+		expected = append(expected, watchResponsePair{
+			Type: client.PeersChanged,
+			Peers: map[string]types.Presence{
+				c1.ID().String(): c1.Presence(),
+			},
+		})
+		cancel2()
```
[the remark this answers is not in the document]
Comment (chacha912) [reply test/integration/peer_awareness_test.go:198]: To check whether events are received when c2's watch is disconnected after canceling, `defer` is not used at that point.

**Pull request merged**